docs/spec/supervisor/supervisor.md

```diff
@@ -0,0 +1,153 @@
+
+```rust
+/// Stores the Light Blocks which could not be confirmed to have originated from
+/// the same chain as the predecesor light blocks.
+type FetchedStack = Stack<LightBlock>;
+
+/// Pair of light block and height of the light block which was highest 
+/// highest one for which we believe originated from the same chain
+type ChainPair = (LightBlock, Height);
+
+/// Traces the history of verify_to_target.
+trait Trace {
+    /// Returns the height of the current highest inserted
+    fn current_height(&self) -> Height;
```

Comment:
What's the use-case for just getting the current height of the `Trace`, can't you obtain that through the `current` `LightBlock`?

Reply:
Just as a convenience method, can be removed.